string[] names
---
int16 success